c_src/lazy_html.cpp

```diff
@@ -714,6 +715,25 @@ ExLazyHTML child_nodes(ErlNifEnv *env, ExLazyHTML ex_lazy_html) {
 
 FINE_NIF(child_nodes, 0);
 
+ExLazyHTML parent_node(ErlNifEnv *env, ExLazyHTML ex_lazy_html) {
+  auto nodes = std::vector<lxb_dom_node_t *>();
+  auto inserted_nodes = std::set<lxb_dom_node_t *>();
```

Comment:
Let's use `std::unordered_set` which where insert and membership check is `O(1)` on average.